lib/events/dynamoevents/dynamoevents.go

```diff
@@ -304,70 +300,163 @@ func New(ctx context.Context, cfg Config) (*Log, error) {
 	}
 	b.svc = svc
 
+	if err := b.configureTable(ctx); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return b, nil
+}
+
+type tableStatus int
+
+const (
+	tableStatusError = iota
+	tableStatusMissing
+	tableStatusNeedsMigration
+	tableStatusOK
+)
+
+func (l *Log) configureTable(ctx context.Context) error {
 	// check if the table exists?
-	ts, err := b.getTableStatus(ctx, b.Tablename)
+	ts, err := l.getTableStatus(ctx, l.Tablename)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
 	switch ts {
 	case tableStatusOK:
 		break
 	case tableStatusMissing:
-		err = b.createTable(ctx, b.Tablename)
+		err = l.createTable(ctx, l.Tablename)
 	case tableStatusNeedsMigration:
-		return nil, trace.BadParameter("unsupported schema")
+		return trace.BadParameter("unsupported schema")
 	}
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
-	err = dynamo.TurnOnTimeToLive(ctx, b.svc, b.Tablename, keyExpires)
+	tableName := aws.String(l.Tablename)
+	ttlStatus, err := l.svc.DescribeTimeToLiveWithContext(ctx, &dynamodb.DescribeTimeToLiveInput{
+		TableName: tableName,
+	})
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(convertError(err))
+	}
+	switch aws.StringValue(ttlStatus.TimeToLiveDescription.TimeToLiveStatus) {
+	case dynamodb.TimeToLiveStatusEnabled, dynamodb.TimeToLiveStatusEnabling:
+	default:
+		_, err = l.svc.UpdateTimeToLiveWithContext(ctx, &dynamodb.UpdateTimeToLiveInput{
+			TableName: tableName,
+			TimeToLiveSpecification: &dynamodb.TimeToLiveSpecification{
+				AttributeName: aws.String("Expires"),
```

Comment:
Previously we had this field as a const.
I think we should keep it as a const but with even greater visibility.
Maybe at a place where we keep consts related to backend storage and/or events.